integration-tests/src/test/java/gov/uk/di/ipv/cri/common/api/util/IpvCoreStubUtil.java

```diff
@@ -0,0 +1,100 @@
+package gov.uk.di.ipv.cri.common.api.util;
+
+import org.apache.http.client.utils.URIBuilder;
+
+import java.io.IOException;
+import java.net.Authenticator;
+import java.net.PasswordAuthentication;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Objects;
+import java.util.Optional;
+
+public class IpvCoreStubUtil {
+
+    private static final String ADDRESS_CRI_DEV = "address-cri-dev";
+
+    public static String getPrivateAPIEndpoint() {
+        return getApiEndpoint(
+                "API_GATEWAY_ID_PRIVATE", "Environment variable PRIVATE API endpoint is not set");
+    }
+
+    private static String getApiEndpoint(String apikey, String message) {
+        String apiEndpoint = System.getenv(apikey);
+        Optional.ofNullable(apiEndpoint).orElseThrow(() -> new IllegalArgumentException(message));
```

Comment:
remove the second `message` argument from this method.
In the IllegalArgumentException constructor pass in something along the lines of 
String.format("Environment variable %s is not assigned", apikey)